File: netty/src/main/java/cn/elmi/sample/netty/TestServerInitializer.java
package cn.elmi.sample.netty;

import io.netty.channel.ChannelInitializer;
import io.netty.channel.ChannelPipeline;
import io.netty.channel.socket.SocketChannel;
import io.netty.handler.codec.http.HttpServerCodec;
import lombok.extern.slf4j.Slf4j;

/**
 * @version 1.0
 * @authur Arthur
 */
@Slf4j
public class TestServerInitializer extends ChannelInitializer<SocketChannel> {

    @Override
    protected void initChannel(SocketChannel ch) throws Exception {
        log.debug("TestServerInitializer.initChannel");
        ChannelPipeline pipeline = ch.pipeline();
        pipeline.addLast("http-server-codec", new HttpServerCodec());

    }

}

File: netty/src/main/kotlin/cn/elmi/sample/netty/TestServerInitializer.kt
package cn.elmi.sample.netty

import io.netty.channel.ChannelInitializer
import io.netty.channel.socket.SocketChannel
import io.netty.handler.codec.http.HttpServerCodec

/**
 * @version 1.0
 * @authur Arthur
 */
class TestServerInitializer : ChannelInitializer<SocketChannel>() {
    @Throws(Exception::class)
    override fun initChannel(ch: SocketChannel) {
//        log.debug("TestServerInitializer.initChannel")
        val pipeline = ch.pipeline()
        pipeline.addLast("http-server-codec", HttpServerCodec())
    }
}